client/rest-high-level/src/test/java/org/elasticsearch/client/IndicesClientIT.java

```diff
@@ -1201,4 +1205,31 @@ public void testGetIndexTemplate() throws Exception {
             new GetIndexTemplatesRequest().names("the-template-*"), client.indices()::getTemplate, client.indices()::getTemplateAsync));
         assertThat(notFound.status(), equalTo(RestStatus.NOT_FOUND));
     }
+
+    public void testPutStoredScript() throws Exception {
+        RestHighLevelClient client = highLevelClient();
+        XContentType xContentType = randomFrom(XContentType.values());
+        PutStoredScriptRequest request = new PutStoredScriptRequest()
+            .id("script1");
+
+        try (XContentBuilder builder = XContentBuilder.builder(xContentType.xContent())) {
```

Comment:
It would be great to encapsulate all that in `PutStoredScriptRequest.toXContent(XContentBuilder builder, Params params)`